monai/losses/dice.py

```diff
@@ -69,12 +69,93 @@ def forward(self, pred, ground, smooth=1e-5):
         assert tsum.shape == psum.shape, ("Ground truth one-hot has differing shape (%r) from source (%r)" %
                                           (tsum.shape, psum.shape))
 
-        batchsize = ground.size(0)
-        tsum = tsum.float().view(batchsize, -1)
-        psum = psum.view(batchsize, -1)
+        batchsize, n_classes = tsum.shape[:2]
+        tsum = tsum.float().view(batchsize, n_classes, -1)
+        psum = psum.view(batchsize, n_classes, -1)
 
         intersection = psum * tsum
         sums = psum + tsum
 
-        score = 2.0 * (intersection.sum(1) + smooth) / (sums.sum(1) + smooth)
-        return 1 - score.sum() / batchsize
+        score = 2.0 * (intersection.sum(2) + smooth) / (sums.sum(2) + smooth)
```

Comment:
Should be `score = 2.0 * intersection.sum(2) / (sums.sum(2) + smooth)` if we're removing the smoothing value from the numerator.

Reply:
this ticket says no... https://github.com/NifTK/NiftyNet/issues/249